yt/frontends/stream/tests/test_stream_particles.py

```diff
@@ -126,3 +146,164 @@ def test_stream_particles():
     assert amr2._get_field_info("all", "particle_position_z").particle_type
     assert amr2._get_field_info("all", "particle_mass").particle_type
     assert not amr2._get_field_info("gas", "density").particle_type
+
+    # Now perform similar checks, but with multiple particle types
+
+    num_dm_particles = 30000
+    xd = np.random.uniform(size=num_dm_particles)
+    yd = np.random.uniform(size=num_dm_particles)
+    zd = np.random.uniform(size=num_dm_particles)
+    md = np.ones(num_dm_particles)
+
+    num_star_particles = 20000
+    xs = np.random.uniform(size=num_star_particles)
+    ys = np.random.uniform(size=num_star_particles)
+    zs = np.random.uniform(size=num_star_particles)
+    ms = 2.0*np.ones(num_star_particles)
+
+    dens = np.random.random(domain_dims)
+
+    fields3 = {"density": dens,
+               ("dm", "particle_position_x"): xd,
+               ("dm", "particle_position_y"): yd,
+               ("dm", "particle_position_z"): zd,
+               ("dm", "particle_mass"): md,
+               ("star", "particle_position_x"): xs,
+               ("star", "particle_position_y"): ys,
+               ("star", "particle_position_z"): zs,
+               ("star", "particle_mass"): ms,
+               "number_of_particles": num_dm_particles+num_star_particles}
```

Comment:
OK, fair enough. Thanks for giving it some more thought. I think you've convinced me.

The two test failures are spurious and I've re-triggered the builds. I'm working on fixing the random Travis and appveyor failures today.